### PR TITLE
Crash fixes and random odds n ends

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] protect listtracking nulls
- [x] listen to crash / unresponsive / uncaught errors
- [x] update protocol to add ? to flow for arrays
- [x] add $Shape to rpcs
- [x] remove empty incomingCallMaps
- [x] handle reject in loadSummaries and etc
- [x] use exact instead of shape

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready to be looked at

File: shared/actions/login/index.js [hunk 1]
```diff
@@ -491,11 +475,12 @@ function makeKex2IncomingMap (dispatch, getState, onBack: SimpleCB, onProvisione
           onBack={() => onBack(response)} />))
     },
     'keybase.1.provisionUi.chooseDevice': ({devices}, response) => {
+      const d = devices || []
       appendRouteElement((
         <SelectOtherDevice
           devices={devices}
           onSelect={deviceID => {
-            const type: DeviceRole = devices[devices.findIndex(d => d.deviceID === deviceID)].type
+            const type: DeviceRole = d[d.findIndex(d => d.deviceID === deviceID)].type
```
Comment (MarcoPolo) [inline shared/actions/login/index.js]: Why not 
```
(devices || []).find(d => d.deviceID === deviceID).type
```
Also what do you do when the deviceID isn't in the array?

**Comment by MarcoPolo** [comment]
👍

File: shared/actions/login/index.js (hunk 1 above)
Comment (chrisnojima) [inline shared/actions/login/index.js]: we don't handle that well now. unlikely to be an issue as you've clicked an item in the list and we're just plucking extra data out

**Pull request merged**